plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java

```diff
@@ -153,213 +195,233 @@ public LocationMode getLocationMode() {
 
     @Override
     public BlobContainer blobContainer(BlobPath path) {
-        return new AzureBlobContainer(path, this, threadPool);
+        return new AzureBlobContainer(path, this);
     }
 
     @Override
     public void close() {
     }
 
-    public boolean blobExists(String blob) throws URISyntaxException, StorageException {
-        // Container name must be lower case.
-        final Tuple<CloudBlobClient, Supplier<OperationContext>> client = client();
-        final OperationContext context = hookMetricCollector(client.v2().get(), getMetricsCollector);
-        final CloudBlobContainer blobContainer = client.v1().getContainerReference(container);
-        return SocketAccess.doPrivilegedException(() -> {
-            final CloudBlockBlob azureBlob = blobContainer.getBlockBlobReference(blob);
-            return azureBlob.exists(null, null, context);
-        });
-    }
+    public boolean blobExists(String blob) throws IOException {
+        final BlobServiceClient client = client();
 
-    public void deleteBlob(String blob) throws URISyntaxException, StorageException {
-        final Tuple<CloudBlobClient, Supplier<OperationContext>> client = client();
-        // Container name must be lower case.
-        final CloudBlobContainer blobContainer = client.v1().getContainerReference(container);
-        logger.trace(() -> new ParameterizedMessage("delete blob for container [{}], blob [{}]", container, blob));
-        SocketAccess.doPrivilegedVoidException(() -> {
-            final CloudBlockBlob azureBlob = blobContainer.getBlockBlobReference(blob);
-            logger.trace(() -> new ParameterizedMessage("container [{}]: blob [{}] found. removing.", container, blob));
-            azureBlob.delete(DeleteSnapshotsOption.NONE, null, null, client.v2().get());
-        });
+        try {
+            Boolean blobExists = SocketAccess.doPrivilegedException(() -> {
+                final BlobClient azureBlob = client.getBlobContainerClient(container).getBlobClient(blob);
+                return azureBlob.exists();
+            });
+            return Boolean.TRUE.equals(blobExists);
+        } catch (Exception e) {
+            logger.warn("can not access [{}] in container {{}}: {}", blob, container, e.getMessage());
```

Comment:
You can make this `trace` again, the exception will bubble up and will be logged at warn level upstream anyway

Reply:
Fixed in https://github.com/elastic/elasticsearch/pull/65140/commits/2c3eefa1117b3371a21f42ea7902058f274fb746